src/Orleans.Core/Manifest/ClientClusterManifestProvider.cs

```diff
@@ -94,6 +94,11 @@ private async Task RunAsync()
                     if (gateway is null || !_gatewayManager.IsGatewayAvailable(gateway))
                     {
                         gateway = _gatewayManager.GetLiveGateway();
+                        if (gateway is null)
+                        {
+                            continue;
```

Comment:
Am I misreading or will this spin, calling `GetLiveGateway` in a tight loop? Perhaps this should contain a delay, similar to the other `continue` statements:

```suggestion
                            // Wait for a gateway to become available.
                            await Task.WhenAny(cancellationTask, Task.Delay(_typeManagementOptions.TypeMapRefreshInterval));
                            continue;
```

Reply:
absolutely, updated accordingly